React Native/Styling.md

```diff
@@ -0,0 +1,211 @@
+
+There are almost no differences between them, but we decided that `styled-components` should be the option when starting a new project. Main reason for that is theming typings which can easily be defined when using `styled-components` while on the other hand `emotion` library doesn't have that clean solution yet.
+
+**Note:** `emotion`  had announced that theming will get better typings in version 11.
+
+Next paragraphs inside the section are related to configuration & usage of `styled-components` library.
+
+#### INSTALLATION
+`npm install --save styled-components` or `yarn add styled-components`
+
+#### USAGE
+To use `styled-components` inside RN app make sure to import `styled-components/native` package as it provides all styling rules needed for RN.
+
+```javascript
+import React from 'react'
+import styled from 'styled-components/native'
+
+const StyledView = styled.View`
+  background-color: papayawhip;
+`
+
+const StyledText = styled.Text`
+  color: palevioletred;
+`
+
+const MyReactNativeComponent: React.FunctionComponent = () => {
+  render() {
+    return (
+      <CircleView>
+        <StyledText>Hello World!</StyledText>
+      </CircleView>
+    )
+  }
+}
+```
+
+If you want to style your own (custom) components using the `styled-components` there are 2 main steps:
+1. Top element of your component needs implement / use `style` props and accept it's value from component prop.
+
+```javascript
+import React from 'react'
+import { ViewStyle } from 'react-native';
+import styled from 'styled-components/native'
+
+interface ICircleProps {
+	style?: ViewStyle | Array<ViewStyle>;
+	radius: number;
+}
+
+const CircleWrapper = styled.View`
+  background-color: papayawhip;
+  height: ${({ radius }) => radius * 2};
+  width: ${({ radius }) => radius * 2};
+  border-radius: ${({ radius }) => radius};
+`
+
+export const Circle: React.FunctionComponent = ({ children, style, radius }) => {
+  render() {
+    return (
+      <CircleWrapper style={style} radius={radius}>
+        {children}
+      </CircleWrapper>
+    )
+  }
+}
+```
+
+2. When you use component (`Circle`) inside your code you can override it's style with:
+
+```javascript
+import React from 'react'
+import { ViewStyle } from 'react-native';
+import { Circle } from '../components/Circle'; // path to Circle component
+import styled from 'styled-components/native'
+
+const AppCircle = styled(Circle)`
+  background-color: tomato;
+`
+
+const App: React.FunctionComponent = () => {
+  render() {
+    return (
+      <AppCircle radius={50}>
+        {children}
+      </AppCircle>
+    )
+  }
+}
+```
+This means that when you're defining your styles inside `AppCircle`,  you're actually creating a normal `Circle` component, that has your styles attached to it. Important thing to mention here is that `AppCircle` inherits all the props from `Circle` component.
+
+When defining styled component in RN you can provide style properties directly:
```

Comment:
Maybe we should add when it's ok to use this? Seems practical for some dynamic styling, but we wouldn't want someone to just create an empty styled component and then add styles inline.

Reply:
I've added simple explanation below code example.